spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelFactory.java

```diff
@@ -0,0 +1,266 @@
+import org.springframework.cloud.stream.config.BindingProperties;
+import org.springframework.cloud.stream.config.ChannelBindingServiceProperties;
+import org.springframework.cloud.stream.converter.AbstractFromMessageConverter;
+import org.springframework.cloud.stream.converter.ByteArrayToStringMessageConverter;
+import org.springframework.cloud.stream.converter.CompositeMessageConverterFactory;
+import org.springframework.cloud.stream.converter.JavaToSerializedMessageConverter;
+import org.springframework.cloud.stream.converter.JsonToPojoMessageConverter;
+import org.springframework.cloud.stream.converter.JsonToTupleMessageConverter;
+import org.springframework.cloud.stream.converter.MessageConverterUtils;
+import org.springframework.cloud.stream.converter.PojoToJsonMessageConverter;
+import org.springframework.cloud.stream.converter.PojoToStringMessageConverter;
+import org.springframework.cloud.stream.converter.SerializedToJavaMessageConverter;
+import org.springframework.cloud.stream.converter.StringToByteArrayMessageConverter;
+import org.springframework.cloud.stream.converter.TupleToJsonMessageConverter;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.integration.channel.AbstractMessageChannel;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.config.ConsumerEndpointFactoryBean;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.messaging.converter.MessageConverter;
+import org.springframework.scheduling.support.PeriodicTrigger;
+import org.springframework.util.Assert;
+import org.springframework.util.MimeType;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * Class that {@link BindableProxyFactory} uses to create message channels.
+ *
+ * @author Marius Bogoevici
+ * @author David Syer
+ * @author Ilayaperumal Gopinathan
+ */
+public class ChannelFactory implements BeanFactoryAware, InitializingBean {
+
+	public static final String SPRING_CLOUD_STREAM_INTERNAL_PREFIX = "spring.cloud.stream.internal";
+
+	public static final String CHANNEL_NAMESPACE_PROPERTY_NAME = SPRING_CLOUD_STREAM_INTERNAL_PREFIX + ".channelNamespace";
+
+	public static final String POLLABLE_BRIDGE_INTERVAL_PROPERTY_NAME = SPRING_CLOUD_STREAM_INTERNAL_PREFIX + ".pollableBridge.interval";
+
+	@Value("${" + CHANNEL_NAMESPACE_PROPERTY_NAME + ":}")
+	private String channelNamespace;
+
+	@Value("${" + POLLABLE_BRIDGE_INTERVAL_PROPERTY_NAME + ":1000}")
+	private int pollableBridgeDefaultFrequency;
+
+	private ConfigurableListableBeanFactory beanFactory;
+
+	@Autowired(required = false)
+	private SharedChannelRegistry sharedChannelRegistry;
```

Comment:
IMO this could go back to BindableChannelProxy, who would consult it and delegate to ChannelFactory if and only if it needs to create a new channel.